### PR TITLE
Fix #17506 - r2 http://example.org works again ##io

### LINKED ISSUE #17506: Can not open file via http
### Work environment

| Questions                                            | Answers
|------------------------------------------------------|--------------------
| OS/arch/bits (mandatory)                             | Ubuntu x86 64
| r2 -v full output, **not truncated** (mandatory)     | radare2 4.6.0-git 25587 @ linux-x86-64 git.4.4.0-574-g3c8890013 commit: 3c889001347fb764db98594b97e65b4e22cf9600 build: 2020-08-21__14:19:08

### Expected behavior
r2 opens file via http

### Steps to reproduce the behavior 
```
r2 http://host/path
[r] Cannot open '...
```

### PR DESCRIPTION
**Pull request by trufae**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Detailed description**

...

**Test plan**

...

**Closing issues**

...

### CODE REVIEW

**Comment by XVilka** [comment]
Breaks the [test on Windows](https://ci.appveyor.com/project/radareorg/radare2/builds/34827477/job/ttm9ao30172yauuw#L6497):
```
[XX] C:\projects\radare2\test\db\io\http r2 http://example.org
ANSICON=1 R2_NOPLUGINS=1 radare2 -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Qc 'r
b 64
ps
' http://example.org
-- stdout
@@ -1,3 +1,6 @@
 1256
-    <meta charset="utf-8" />
-    <meta http-equiv="Content-type"
+<!doctype html>
+<html>
+<head>
+    <title>Example Domain</title>
+
[**]                
```

File: libr/cons/input.c [hunk 1]
```diff
@@ -143,6 +143,7 @@ R_API int r_cons_arrow_to_hjkl(int ch) {
 	}
 	switch (ch) {
 	case 0x1b:
+eprintf ("thats a quit\n");
```
Comment (ret2libc) [inline libr/cons/input.c:146]: please remove these before merging.

**Review by ret2libc — CHANGES_REQUESTED** [review]
Just to ensure debugging printf are removed before merging.

File: libr/socket/socket.c [hunk 2]
```diff
@@ -772,7 +771,7 @@ R_API void r_socket_printf(RSocket *s, const char *fmt, ...) {
 	if (s->fd != R_INVALID_SOCKET) {
 		va_start (ap, fmt);
 		vsnprintf (buf, BUFFER_SIZE, fmt, ap);
-		r_socket_write (s, buf, strlen (buf));
+		(void) r_socket_write (s, buf, strlen (buf));
```
Comment (trufae) [inline libr/socket/socket.c:774]: r_socket_printf should return int to remove this void cast

File: libr/socket/socket.c [hunk 3]
```diff
@@ -311,7 +311,7 @@ R_API bool r_socket_connect(RSocket *s, const char *host, const char *port, int
 					s->fd = -1;
 					continue;
 				}
-				r_socket_block_time (s, 0, 0, 0);
+				r_socket_block_time (s, 10000, 1000, 0);
```
Comment (trufae) [inline libr/socket/socket.c:314]: i would like to have some defines to specify some default timeout values here, (and reuse them in the other cases below). which time do you think it shuold be fine to define for waiting until the connection succeds?

from internet:

* In Windows the value is dynamic for established conections, though the default for initial connections is 72 seconds. 

imho thats too much, i would go for having lower timeouts like 5s

Comment (ret2libc) [reply libr/socket/socket.c:314]: No real preference here. 5s seems like a good random value for now!

**Review by ret2libc — APPROVED** [review]

File: libr/socket/socket.c (hunk 3 above)
Comment (trufae) [reply libr/socket/socket.c]: Don't aprove it until the implementation is done. you can review it again now

**Comment by trufae** [comment]
Screenshot 2020-08-28 at 11 16 15

**Comment by GustavoLCR** [comment]
@trufae This fixes it:
```diff
diff --git a/libr/socket/socket_http.c b/libr/socket/socket_http.c
index 3aaf7f0ee..9b4efba45 100644
--- a/libr/socket/socket_http.c
+++ b/libr/socket/socket_http.c
@@ -34,7 +34,7 @@ static char *socket_http_answer(RSocket *s, int *code, int *rlen, ut32 redirecti
        r_return_val_if_fail (s, NULL);
        const char *p;
        int ret, len = 0, delta = 0;
-       char *dn;
+       char *dn = NULL;
        RBuffer *b = r_buf_new ();
        if (!b) {
                return NULL;
@@ -47,11 +47,21 @@ static char *socket_http_answer(RSocket *s, int *code, int *rlen, ut32 redirecti
        }
        r_buf_read_at (b, 0, (ut8 *)buf, olen);
        buf[olen] = 0;
-       if ((dn = (char*)r_str_casestr (buf, "\n\n"))) {
-               delta += 2;
-       } else if ((dn = (char*)r_str_casestr (buf, "\r\n\r\n"))) {
-               delta += 4;
+       char *dnn = (char *)r_str_casestr (buf, "\n\n");
+       char *drn = (char *)r_str_casestr (buf, "\r\n\r\n");
+       if (dnn) {
+               if (drn && (drn < dnn)) {
+                       dn = drn;
+                       delta = 4;
+               } else {
+                       dn = dnn;
+                       delta = 2;
+               }
        } else {
+               dn = drn;
+               delta = 4;
+       }
+       if (!dn) {
                goto exit;
        }

diff --git a/test/db/io/http b/test/db/io/http
index 9ba707cb0..ea3f5776b 100644
--- a/test/db/io/http
+++ b/test/db/io/http
@@ -7,8 +7,11 @@ ps
 EOF
 EXPECT=<<EOF
 1256
-    <meta charset="utf-8" />
-    <meta http-equiv="Content-type"
+<!doctype html>
+<html>
+<head>
+    <title>Example Domain</title>
+
 EOF
 RUN

```

**Comment by radare** [comment]
Why red if all CI is green? windows test fixed by @GustavoLCR pr, so imho its good to merge. pls review

**Pull request merged**